config/formats.js

```diff
@@ -496,79 +496,138 @@ let Formats = [
+			`&bullet; <a href="https://www.smogon.com/forums/threads/3618488/">Partners in Crime</a>`,
 		],
 
-		mod: 'gen7',
-		ruleset: ['[Gen 7] OU'],
-		banlist: ['Blissey'],
-		onBeforeFaint: function (pokemon) {
-			let prevo = pokemon.baseTemplate.isMega ? this.getTemplate(pokemon.baseTemplate.baseSpecies).prevo : pokemon.baseTemplate.prevo;
-			if (!prevo || pokemon.set.ability === 'Battle Bond') return;
-			let template = this.getTemplate(pokemon.set.species);
+		mod: 'pic',
+		gameType: 'doubles',
+		ruleset: ['[Gen 7] Doubles OU', 'Sleep Clause Mod'],
+		banlist: ['Huge Power', 'Imposter', 'Parental Bond', 'Pure Power', 'Wonder Guard', 'Mimic', 'Sketch', 'Transform'],
+		onSwitchInPriority: 2,
+		onSwitchIn: function (pokemon) {
+			if (this.p1.active.every(ally => ally && !ally.fainted)) {
+				let p1a = this.p1.active[0], p1b = this.p1.active[1];
+				if (p1a.ability !== p1b.ability) {
+					let p1aInnate = 'ability' + p1b.ability;
+					p1a.volatiles[p1aInnate] = {id: p1aInnate, target: p1a};
+					let p1bInnate = 'ability' + p1a.ability;
+					p1b.volatiles[p1bInnate] = {id: p1bInnate, target: p1b};
+				}
+			}
+			if (this.p2.active.every(ally => ally && !ally.fainted)) {
+				let p2a = this.p2.active[0], p2b = this.p2.active[1];
+				if (p2a.ability !== p2b.ability) {
+					let p2a_innate = 'ability' + p2b.ability;
+					p2a.volatiles[p2a_innate] = {id: p2a_innate, target: p2a};
+					let p2b_innate = 'ability' + p2a.ability;
+					p2b.volatiles[p2b_innate] = {id: p2b_innate, target: p2b};
+				}
+			}
+			let ally = pokemon.side.active.find(ally => ally && ally !== pokemon && !ally.fainted);
+			if (ally && ally.ability !== pokemon.ability) {
+				// @ts-ignore
+				if (!pokemon.innate) {
+					// @ts-ignore
+					pokemon.innate = 'ability' + ally.ability;
+					// @ts-ignore
+					delete pokemon.volatiles[pokemon.innate];
+					// @ts-ignore
+					pokemon.addVolatile(pokemon.innate);
+				}
+				// @ts-ignore
+				if (!ally.innate) {
+					// @ts-ignore
+					ally.innate = 'ability' + pokemon.ability;
+					// @ts-ignore
+					delete ally.volatiles[ally.innate];
+					// @ts-ignore
+					ally.addVolatile(ally.innate);
+				}
+			}
+		},
+		onSwitchOut: function (pokemon) {
+			// @ts-ignore
+			if (pokemon.innate) {
+				// @ts-ignore
+				pokemon.removeVolatile(pokemon.innate);
+				// @ts-ignore
+				delete pokemon.innate;
+			}
+			let ally = pokemon.side.active.find(ally => ally && ally !== pokemon && !ally.fainted);
 			// @ts-ignore
-			let abilitySlot = Object.keys(template.abilities).find(slot => template.abilities[slot] === pokemon.set.ability);
-			template = this.getTemplate(prevo);
+			if (ally && ally.innate) {
+				// @ts-ignore
+				ally.removeVolatile(ally.innate);
+				// @ts-ignore
+				delete ally.innate;
+			}
+		},
+		onFaint: function (pokemon) {
+			// this.add('-hint', 'checking ally ability after faint of ' + pokemon);
```

Comment:
Oops, looks like I have three lines of old debugging code that I need to delete...